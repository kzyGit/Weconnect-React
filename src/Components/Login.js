import React, { Component } from 'react';
import '../Styles/App.css';
import Header,{Base_url} from './Header';
import Footer from './Footer';
import axios from 'axios';
import swal from 'sweetalert';
import {browserHistory} from 'react-router';
import { Link } from 'react-router';

class Login extends Component {
  render() {
    return (
      <div className="row">
      <Header />
      <LoginContent />
      <Footer />
      </div>     
    );
  }
}

class LoginContent extends Component {
  state = {
    login:[]
  }
 
  login = (e) => {
    e.preventDefault();
    const username = e.target.elements.username.value;
    const password = e.target.elements.password.value;
    localStorage.setItem('loggedIn', false)

    axios.post(`${Base_url}/auth/login`, {
      username: username,
      password: password
    }).then(response => {

      if(response.data.status_code === 204){
        swal('User not found, kindly use a registered username');
        localStorage.removeItem('loggedIn')
        }
      else{
        
      const access_token = response.data.access_token
      const username = response.data.username
      const email = response.data.email
      

      localStorage.setItem('loggedIn', true)

      localStorage.setItem("access_token", access_token)
      localStorage.setItem("username", username)
      localStorage.setItem("email", email)
      browserHistory.push('/dashboard')
      swal({
          title: "Success!",
          text: "You have successfully logged in",
          icon: "success",
          button: "Ok",
        });
  }})
  .catch(error => {
    if(error.response.status === 401){
      const message = error.response.data.Error
      localStorage.removeItem('loggedIn')
      swal("Error!!", message, "error");
      
      }

  });
}
  render() {
    return (
      <div className="signupcontent">
        <div className = "row">
          <h3 style={{paddingLeft:'20px',color:'break'}}>Login</h3><br />
          <div className="col-md-1" ></div>
          <div className="col-md-10" >
            
            <form onSubmit={this.login}>
              <div className="form-group">
                <label>Username:</label><input type="text" name='username' className="field" />
                
              </div><br />
              <div className="form-group">
                <label >Password:</label>
                <input type="password" className="field" name='password'/>
              </div><br />
              <button style={{float:'right'}} type="submit" className="btn btn-default">Submit</button>
          </form>
              
  
          </div>

          <br /><br />
          <Link to={'/resetpassword'}><span style={{ textAlign:'center'}}>Forgot password?</span></Link>

          <Link to={'/signup'}><span style={{ paddingLeft:'15%'}}>No account? Signup</span></Link>


          <div className = "col-md-1"></div>
  
        </div>
      </div>
    
          );
        }
      }

export default Login;
