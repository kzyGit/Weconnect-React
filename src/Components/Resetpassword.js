import React, { Component } from 'react';
import '../Styles/App.css';
import Header, { Base_url } from './Header';
import Footer from './Footer';
import swal from 'sweetalert';
import axios from 'axios'
import { browserHistory } from 'react-router';

/**
 * Component that enables a user to request for password reset
 */
class ResetPassword extends Component {
    render() {
        return (
            <div className="row">
                <Header />
                <div className="signupcontent">
                    <div className="row">
                        <h3 style={{ paddingLeft: '20px', color: 'break' }}>Request Password Reset:</h3><br />
                        <div className="col-md-1" ></div>
                        <div className="col-md-10" >
                            <Request requestResetPass={this.requestResetPass} />
                        </div>
                        <div className="col-md-1"></div>
                    </div>
                </div>
                <Footer />
            </div>
        );
    }
}

class Request extends Component {
    render() {
        return (
            <form className="reset-pwd-form" onSubmit={this.requestResetPass}>
                <div className="form-group">
                    <label>Email:</label>
                    <input type="email" className="field" name='email' />
                </div><br />
                <a style={{ float: 'right' }}><button type="submit" className="btn btn-primary">Send Request</button></a><br />
            </form>
        );
    }

    /**
     * Makes server request to validate email and send a reset link to user's email
     */
    requestResetPass = (e) => {
        e.preventDefault()
        const email = e.target.elements.email.value;

        axios.post(`${Base_url}/auth/reset_password`, {
            email: email
        }).then(response => {

            if (response.data.status_code === 204) {
                swal("Error!!", "Unrecognised email, kindly ensure to use the email you registered with", "error");
            }
            else {
                browserHistory.push('/login')
                swal({
                    title: "Success!",
                    text: "Kindly check your email for a token to reset your password",
                    icon: "success",
                    button: "OK",
                });
            }
        }).catch(error => {
            if (error.response.status === 400) {
                swal("Error!!", error.response.data.Error, "error");
            }
        });
    }
}

export default ResetPassword;
export {
    Request
}