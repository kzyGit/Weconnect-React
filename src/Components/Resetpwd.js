import React, { Component } from 'react';
import '../Styles/App.css';
import Header from './Header';
import Footer from './Footer';
import swal from 'sweetalert';
import axios from 'axios'
import PropTypes from 'prop-types'
// import { browserHistory } from 'react-router';


class EditPassword extends Component {

    constructor(props) {
        super(props)
        this.state = {
        };
    }

    requestResetPass = (e) => {
        e.preventDefault()

        const auth_token = this.props.params.token;
        const config = { headers: { 'Authorization': "bearer " + auth_token } }
        
        const new_password = e.target.elements.new_pwd.value;
        const confirm_password = e.target.elements.confirm_pwd.value;

        axios.put('https://weconnectapi-v2.herokuapp.com/api/v1/auth/reset_password', config, {
            new_password: new_password,
            confirm_password: confirm_password
        }).then(response => {

            swal({
                title: "Success!",
                text: response.data.Success,
                icon: "success",
                button: "OK",
            });

        }).catch(error => {

            if (error.response.status === 400) {
                swal("Error!!", error.response.data.Error, "error");
            }
            else if (error.response.status === 401) {
                swal("Error!!", error.response.data.Error, "error");
            }
        });

    }

    render() {
        return (
            <div className="row">
                <Header />


                <div className="signupcontent">
                    <div className="row">

                        <h3 style={{ paddingLeft: '20px', color: 'break' }}>Reset Password:</h3><br />

                        <div className="col-md-1" ></div>
                        <div className="col-md-10" >

                            <form className="reset-pwd-form" onSubmit={this.requestResetPass}>
                                <div className="form-group">
                                    <label>New Password:</label>
                                    <input type="password" className="field" name='new_pwd' />


                                </div>
                                <br />
                                <div className="form-group">
                                    <label>Confirm Password:</label>
                                    <input type="password" className="field" name='confirm_pwd' />

                                </div>
                                <br />

                                <a style={{ float: 'right' }}><button type="submit" className="btn btn-primary">Reset</button></a>
                                <br />
                            </form>


                        </div>
                        <div className="col-md-1"></div>

                    </div>
                </div>
                <Footer />
            </div>
        );
    }
}

EditPassword.propTypes = {
    params:PropTypes.object.isRequired
  
  }
export default EditPassword;