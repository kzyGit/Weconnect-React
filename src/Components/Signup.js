import React, { Component } from 'react';
import '../Styles/App.css';
import Header from './Header';
import Footer from './Footer';
import axios from 'axios'
import swal from 'sweetalert';
import {browserHistory} from 'react-router';

class Signup extends Component {
  render() {
    return (
      <div className="row">
      <Header />
      <SignupContent />
      <Footer />
      </div>     
    );
  }
}

class SignupContent extends Component {

  signUp = (e) => {
    e.preventDefault()

    const username =  e.target.elements.username.value;
    const email =  e.target.elements.email.value;
    const password =  e.target.elements.password.value;
    // eslint-disable-next-line
    const confirm_password =  e.target.elements.confirm_password.value;

    axios.post('https://weconnectapi-v2.herokuapp.com/api/v1/auth/register', {
      username: username,
      email: email,
      password: password,
      confirm_password: confirm_password

    }).then(response =>{
      browserHistory.push('/login')
      // sweet alert pop up
      swal({
          title: "Success!",
          text: "You have registered successfully",
          icon: "success",
          button: "Log in",
        });
  })
  .catch(error => {
      
      if (error.response.status === 409){
          const message = error.response.data.Error
          swal("Error!!", message, "error");
      }
      else if(error.response.status === 400){
          const message = error.response.data.Error
          swal("Error!!", message, "error");
      }
  });
  }
  render() {
    return (
      <div className="signupcontent">
        <div className = "row">
          <h3 style={{paddingLeft:'20px',color:'break'}}>Signup</h3><br />
          <div className="col-md-1" ></div>
          <div className="col-md-10" >
            
            <form onSubmit={this.signUp}>
              <div className="form-group">
                <label>Username:</label><input type="text" className="field" name="username"  required/>
                
              </div><br />
              <div className="form-group">
                <label>Email:</label>
                <input type="email" className="field" name='email' required/>
              </div><br />
              <div className="form-group">
                <label>Password:</label>
                <input type="password" className="field" name='password' required/>
              </div><br />
              <div className="form-group"> 
                <label>Confirm pwd:</label>
                <input type="password" className="field" name='confirm_password'/>
              
              </div><br />
            <a href="index.html" style={{float:'right'}}><button type="submit" className="btn btn-default">Submit</button></a>
          </form>
              
  
          </div>
          <div className = "col-md-1"></div>
  
        </div>
      </div>
    
          );
        }
      }

export default Signup;
