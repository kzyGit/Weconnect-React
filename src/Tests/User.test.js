import React from 'react';
import { shallow } from 'enzyme';
import Login from '../Components/Login';
import Signup from '../Components/Signup';
import Resetpassword from '../Components/Resetpassword';
import Resetpwd from '../Components/Resetpwd';
import { shallowToJson } from 'enzyme-to-json';


describe('Login component', () => {

    const localStorageMock = {
        getItem: jest.fn(),
        setItem: jest.fn(),
        clear: jest.fn()
    };
    global.localStorage = localStorageMock;

    const wrapper = shallow(<Login />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('contains Header', () => {
        expect(wrapper.find('Header')).toHaveLength(1);
    });
    it('contains Footer', () => {
        expect(wrapper.find('Footer')).toHaveLength(1);
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(8);
    });

    it('contains headers', () => {
        expect(wrapper.find('h3')).toHaveLength(1);
    });

    it('contains Forms', () => {
        expect(wrapper.find('form')).toHaveLength(1);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });
    it('contains Link', () => {
        expect(wrapper.find('Link')).toHaveLength(2);
    });
});


describe('Signup component', () => {

    const wrapper = shallow(<Signup />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('contains Header', () => {
        expect(wrapper.find('Header')).toHaveLength(1);
    });
    it('contains Footer', () => {
        expect(wrapper.find('Footer')).toHaveLength(1);
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(10);
    });

    it('contains headers', () => {
        expect(wrapper.find('h3')).toHaveLength(1);
    });

    it('contains Forms', () => {
        expect(wrapper.find('form')).toHaveLength(1);
    });

    it('contains button', () => {
        expect(wrapper.find('button')).toHaveLength(1);
    });
});

describe('Resetpassword component', () => {

    const wrapper = shallow(<Resetpassword />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
})

