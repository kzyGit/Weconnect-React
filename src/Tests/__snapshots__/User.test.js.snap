// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders properly 1`] = `
<div>
  <Header />
  <div
    className="signupcontent"
  >
    <div
      className="row"
    >
      <h3
        style={
          Object {
            "color": "break",
            "paddingLeft": "20px",
          }
        }
      >
        Login
      </h3>
      <br />
      <div
        className="col-md-1"
      />
      <div
        className="col-md-10"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label>
              Username:
            </label>
            <input
              className="field"
              name="username"
              type="text"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label>
              Password:
            </label>
            <input
              className="field"
              name="password"
              type="password"
            />
          </div>
          <br />
          <button
            className="btn btn-default"
            style={
              Object {
                "float": "right",
              }
            }
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <br />
      <br />
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/resetpassword"
      >
        <span
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Forgot password?
        </span>
      </Link>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/signup"
      >
        <span
          style={
            Object {
              "paddingLeft": "15%",
            }
          }
        >
          No account? Signup
        </span>
      </Link>
      <div
        className="col-md-1"
      />
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Logout component renders properly 1`] = `
<div
  className="row"
>
  <Header />
  <LogoutContent />
  <Footer />
</div>
`;

exports[`Signup component renders properly 1`] = `
<div
  className="row"
>
  <Header />
  <div
    className="signupcontent"
  >
    <div
      className="row"
    >
      <h3
        style={
          Object {
            "color": "break",
            "paddingLeft": "20px",
          }
        }
      >
        Signup
      </h3>
      <br />
      <div
        className="col-md-1"
      />
      <div
        className="col-md-10"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label>
              Username:
            </label>
            <input
              className="field"
              name="username"
              required={true}
              type="text"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label>
              Email:
            </label>
            <input
              className="field"
              name="email"
              required={true}
              type="email"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label>
              Password:
            </label>
            <input
              className="field"
              name="password"
              required={true}
              type="password"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label>
              Confirm pwd:
            </label>
            <input
              className="field"
              name="confirm_password"
              type="password"
            />
          </div>
          <br />
          <a
            href="index.html"
            style={
              Object {
                "float": "right",
              }
            }
          >
            <button
              className="btn btn-default"
              type="submit"
            >
              Submit
            </button>
          </a>
        </form>
      </div>
      <div
        className="col-md-1"
      />
    </div>
  </div>
  <Footer />
</div>
`;
